splunk-otel-android-volley/src/main/java/com/splunk/rum/VolleyHttpClientAttributesGetter.java

```diff
@@ -68,18 +71,26 @@ public String method(RequestWrapper requestWrapper) {
 
     @Override
     public List<String> requestHeader(RequestWrapper requestWrapper, String name) {
-        String header;
+        Request<?> request = requestWrapper.getRequest();
         try {
-            header = requestWrapper.getRequest().getHeaders().get(name);
-            if (header == null) {
-                header = requestWrapper.getAdditionalHeaders().get(name);
+            String result = findCaseInsensitive(name, request.getHeaders());
```

Comment:
I believe it would be more correct to return all headers that match the given name instead of returning only the first one.

Reply:
Looks like Volley uses `Map<String, String>` to store the request headers -- there's no way we'll ever have more than 1 value for a given name.

Reply:
My understanding is that if you have {"foo":"bar", "FOO":"baz"} then asking for "Foo" should return ["bar", "baz"] because header names should be compared in case insensitive way.

Reply:
@laurit you're right, but as @mateuszrzeszutek pointed out this is a deficiency in the Volley client/request APIs. As far as I can tell, there isn't even any way of specifying multiple values for a single request header, without doing some case sensitive `Map` hackery (eg. `Cookie: abc`, `COokie: def`, ... `COOKIE: zzz`).

The response side does handle multiple values, but it might still suffer from case sensitivity. I'll revisit that and add missing test coverage.

Reply:
Oh, I think I see the point now tho -- if multiple were specified with different casing we would only return the first one. I'll fix it.